SpiralCompiler/CodeGen/Compiler.cs

```diff
@@ -0,0 +1,330 @@
+using SpiralCompiler.Semantics;
+using SpiralCompiler.Syntax;
+
+namespace SpiralCompiler.CodeGen;
+public class Compiler : AstVisitorBase<Operand>
+{
+    private FunctionDef currentFuncDef = null!;
+    private Dictionary<Symbol.Function, FunctionDef> compiledFunctions = new(ReferenceEqualityComparer.Instance);
+    private BasicBlock currentBasicBlock = null!;
+    private int registerIndex = 0;
+    private int labelIndex = 0;
+
+    public override string ToString() => string.Join(Environment.NewLine, compiledFunctions.Values);
```

Comment:
Do you really want to have the compiler itself have this method? Isn't it enough that Module has it?